bot/exts/utils/snekbox.py

```diff
@@ -74,7 +74,13 @@ async def upload_output(self, output: str) -> Optional[str]:
 
     @staticmethod
     def prepare_input(code: str) -> str:
-        """Extract code from the Markdown, format it, and insert it into the code template."""
+        """
+        Extract code from the Markdown, format it, and insert it into the code template.
+
+        If there is Markdown, ignores surrounding text.
+        If there are several Markdown parts in the message, concatenates only the code blocks.
+        If there is inline code but no code blocks, takes the first instance of inline code.
```

Comment:
Calling it "Markdown" is too vague.

```suggestion
        If there is any code block, ignore text outside the code block.
        If there are several fenced code blocks, concatenate only the fenced code blocks.
        Prefer a fenced code block over an inline code block if both are present.
```

Reply:
Hmm, I tried to rely on the terminology in the existing docstring. Would "formatted code" be more accurate?

Reply:
Ah, missed the suggestion

Reply:
Your third line doesn't convey the meaning in my third line. I tried to convey the preference by saying "concatenates only the code blocks", though I can add "and no inline code blocks"

Reply:
Third line? Did you mean second? It says "concatenate only the fenced code blocks" which makes it clear inline code blocks are excluded, because inline ≠ fenced.

Reply:
I mean that the diff has a "-" on the third line "If there is inline code but no code blocks, takes the first instance of inline code" and I don't see an alternative in your suggestion

Reply:
Sorry, I get it now. I updated the suggestion. I suppose your original third line was okay but I feel mine is more succinct.